lib/httplib/csrf/csrf.go

```diff
@@ -40,72 +40,86 @@ const (
 // it implements "double submit cookie" approach to check against CSRF attacks
 // https://www.owasp.org/index.php/Cross-Site_Request_Forgery_%28CSRF%29_Prevention_Cheat_Sheet#Double_Submit_Cookie
 func AddCSRFProtection(w http.ResponseWriter, r *http.Request) (string, error) {
-	encodedToken := ""
-	token, err := extractFromCookie(r)
+	token, err := ExtractTokenFromCookie(r)
 	// if there was an error retrieving the token, the token doesn't exist
 	if err != nil || len(token) == 0 {
-		encodedToken, err = utils.CryptoRandomHex(tokenLenBytes)
+		token, err = utils.CryptoRandomHex(tokenLenBytes)
 		if err != nil {
 			return "", trace.Wrap(err)
 		}
-	} else {
-		encodedToken = hex.EncodeToString(token)
+	}
+	save(token, w)
+	return token, nil
+}
+
+// VerifyHTTPHeader checks if HTTP header value matches the cookie.
+func VerifyHTTPHeader(r *http.Request) error {
+	token := r.Header.Get(HeaderName)
+	if len(token) == 0 {
+		return trace.BadParameter("cannot retrieve CSRF token from HTTP header %q", HeaderName)
 	}
 
-	save(encodedToken, w)
-	return encodedToken, nil
+	err := VerifyToken(token, r)
```

Comment:
this block below can be one-liner: `return trace.Wrap(VerifyToken(token, r))`. No need to check for nils.